src/input/manager.rs

```diff
@@ -1601,150 +1311,130 @@ impl Manager {
-                                let Ok(device) = ::udev::Device::from_subsystem_sysname(
-                                    subsystem.to_string(),
-                                    name.clone(),
-                                ) else {
-                                    log::debug!(
-                                        "Unable to create UdevDevice from {base_path}/{name} to check initialization"
-                                    );
-                                    attempt += 1;
-                                    tokio::time::sleep(Duration::from_millis(10)).await;
-                                    continue;
-                                };
-
-                                if device.is_initialized() {
-                                    break;
-                                }
-                            };
-                            log::trace!("{base_path}/{name} is not yet initialized by udev");
+                            _ => None,
+                        }
+                    };
+                    let Some(subsystem) = subsystem else {
+                        log::trace!("No supported subsystem detected for {base_path}/{name}");
+                        continue;
+                    };
 
-                            tokio::time::sleep(Duration::from_millis(10)).await;
-                            attempt += 1;
+                    // Wait until the device has initialized with udev
+                    const MAX_TRIES: u8 = 80;
+                    let mut attempt: u8 = 0;
+                    loop {
+                        // Break after max attempts reached
+                        if attempt > MAX_TRIES {
+                            log::warn!("Unable to create initialized UdevDevice for {base_path}/{name} after {MAX_TRIES} attempts.");
+                            continue 'outer;
                         }
 
-                        // Create a udev device for the device
-                        let Ok(device) = ::udev::Device::from_subsystem_sysname(
-                            subsystem.to_string(),
-                            name.clone(),
-                        ) else {
-                            log::warn!("Unable to create UdevDevice from {base_path}/{name}");
-                            continue;
+                        // Try to get the device from udev to check its initialization state
+                        {
+                            let Ok(device) = ::udev::Device::from_subsystem_sysname(
+                                subsystem.to_string(),
+                                name.clone(),
+                            ) else {
+                                log::debug!(
+                                    "Unable to create UdevDevice from {base_path}/{name} to check initialization"
+                                );
+                                attempt += 1;
+                                tokio::time::sleep(Duration::from_millis(10)).await;
```

Comment:
I know this was 10ms before, but my Elite 2 controller doesn't initialize fully before the 800ms have passed. It seems to connect fine if this is 50ms.